third_party/move/move-compiler-v2/src/file_format_generator/module_generator.rs

```diff
@@ -217,20 +217,24 @@ impl ModuleGenerator {
         }
         let struct_handle = self.struct_index(ctx, loc, struct_env);
         let fields = struct_env.get_fields();
-        let field_information = FF::StructFieldInformation::Declared(
-            fields
-                .map(|f| {
-                    let field_loc = f.get_loc();
-                    self.source_map
-                        .add_struct_field_mapping(def_idx, ctx.env.to_ir_loc(field_loc))
-                        .expect(SOURCE_MAP_OK);
-                    let name = self.name_index(ctx, field_loc, f.get_name());
-                    let signature =
-                        FF::TypeSignature(self.signature_token(ctx, loc, &f.get_type()));
-                    FF::FieldDefinition { name, signature }
-                })
-                .collect(),
-        );
+        let field_information = if struct_env.get_field_count() == 0 {
+            FF::StructFieldInformation::Native
```

Comment:
Can you explain why Native is the right choice here?  How is it treated differently and where?

We also might want a comment here so this is not undone in the future by someone who doesn't understand why you did it.

Reply:
If a user-defined struct has no field, a `dummy_field` will be added. So if a struct does not have any field, it means it is a native struct. Also, the bytecode verifier will generate the error `ZERO_SIZED_STRUCT` for a struct with no fields if it is not native. I will add a comment here.

Reply:
We should ideally not resort to secondary "happenstance" information (i.e., invariants that hold today, but may not under future refactorings), especially for information that is explicitly specified (all `native` structs have to be explicitly mentioned as such in the code, right)?

For functions, we maintain information about whether is is native. For structs, we should analogously also maintain such information (I am assuming it is not in the move model, which may have just been an oversight). Then, we can use that explicit information here.

Reply:
Thanks @vineethk. I added the field `is_native` to explicitly mark a native struct. PTAL